clang/lib/CodeGen/CodeGenTypes.cpp

```diff
@@ -749,10 +749,6 @@ llvm::StructType *CodeGenTypes::ConvertRecordDeclType(const RecordDecl *RD) {
     return Ty;
   }
 
-  assert((!Context.getLangOpts().SYCLIsDevice || !isa<CXXRecordDecl>(RD) ||
-          !dyn_cast<CXXRecordDecl>(RD)->isPolymorphic()) &&
-         "Types with virtual functions not allowed in SYCL");
-
```

Comment:
Are there any other parts of https://github.com/intel/llvm/commit/08b20c157ce6196f22906039859f6d70da11da4c, which should be reverted?

Reply:
I don't believe so, the SEMA parts were mostly reverted previously.